src/client/datascience/notebook/types.ts

```diff
@@ -13,6 +13,7 @@ export interface INotebookControllerManager {
     // Marked test only, just for tests to access registered controllers
     registeredNotebookControllers(): VSCodeNotebookController[];
     getInteractiveController(): Promise<VSCodeNotebookController | undefined>;
+    getPreferredNotebookController(document: NotebookDocument): VSCodeNotebookController | undefined;
```

Comment:
I don't think this is being used.

Reply:
It's used in the helper.ts file to find the closest match for a kernel.

Reply:
Basically the the tests that used to call waitForKernelToGetAutoSelected are now instead waiting for the preferred kernel to be auto selected (which effectively is the same as what happened before)

Reply:
Yes found that.